crates/lib/src/execution_data.rs

```diff
@@ -352,19 +352,31 @@ mod describe_register_map {
     use maplit::hashmap;
     use ndarray::prelude::*;
 
+    use crate::qpu::result_data::MemoryValues;
     use crate::qpu::QpuResultData;
     use crate::qvm::QvmResultData;
 
     use super::{RegisterData, RegisterMap};
     use qcs_api_client_grpc::models::controller::readout_values::Values;
-    use qcs_api_client_grpc::models::controller::{IntegerReadoutValues, ReadoutValues};
+    use qcs_api_client_grpc::models::controller::{
+        self, DataValue as ControllerMemoryValue, IntegerDataValue, IntegerReadoutValues,
+        ReadoutValues,
+    };
 
     fn dummy_readout_values(v: Vec<i32>) -> ReadoutValues {
         ReadoutValues {
             values: Some(Values::IntegerValues(IntegerReadoutValues { values: v })),
         }
     }
 
+    fn dummy_memory_values(v: Vec<i64>) -> ControllerMemoryValue {
+        ControllerMemoryValue {
+            value: Some(controller::data_value::Value::Integer(IntegerDataValue {
+                data: v,
+            })),
+        }
+    }
+
```

Comment:
Should there be test cases for the other variants (Binary + Real)?

Reply:
Updated the test to include all variants.